RxSwift/Traits/Infallible/Infallible+Concurrency.swift

```diff
@@ -27,11 +27,41 @@ public extension InfallibleType {
                 onCompleted: { continuation.finish() },
                 onDisposed: { continuation.onTermination?(.cancelled) }
             )
-
+            
             continuation.onTermination = { @Sendable _ in
                 disposable.dispose()
             }
         }
     }
+    
+    /**
+     Allows converting asynchronous block to `Infailable` trait.
+     
+     - Parameters:
+        - priority: The priority of the task.
+        - detached: Detach when creating the task.
+        - block: An asynchronous block.
+     - Returns: An Infailable emits value from `block` parameter.
+     */
+    static func from(priority: TaskPriority? = nil, detached: Bool = false, _ block: @escaping () async -> Element) -> Infallible<Element> {
+        return .create { observer in
+            let operation: @Sendable () async -> Void = {
+                let element = await block()
+                observer(.next(element))
+                observer(.completed)
+            }
+            let task: Task<Void, Swift.Error>
+            
+            if detached {
+                task = Task.detached(priority: priority, operation: operation)
+            } else {
+                task = Task(priority: priority, operation: operation)
+            }
+            
+            return Disposables.create {
+                task.cancel()
```

Comment:
~Good call to cancel the tasks, but the `operation` never checks if the task was cancelled. This will only do anything if the code passed in as `block`~

~Would it make sense to add `if Task.isCancelled { return }` after `let element = await block()` and not send anything to `observer`, as if the observable never fired anything?
Or would it make more sense to not send `.next(element)` but to still send `.completed`?~

I left that note to show my train of thought, but now I am thinking it _is_ best to leave this as is and not check `Task.isCancelled` since it is up to the user passing in Block to decide what to do when their Task is cancelled. They still may want to return a partial result perhaps. That should be up to them.

I was thinking there should also be a way to indicate they want to return nothing when their Task gets cancelled, but I think the catch block (in the other three, not Infallible) is probably sufficient for that. If they end user cares to know their Single/Maybe/Completable threw an error because the Task was cancelled and not some other error they can check the error type.

From the [documentation](https://developer.apple.com/documentation/swift/task):
"Tasks include a shared mechanism for indicating cancellation, but not a shared implementation for how to handle cancellation. Depending on the work you’re doing in the task, the correct way to stop that work varies. Likewise, it’s the responsibility of the code running as part of the task to check for cancellation whenever stopping is appropriate. In a long-task that includes multiple pieces, you might need to check for cancellation at several points, and handle cancellation differently at each point. If you only need to throw an error to stop the work, call the Task.checkCancellation() function to check for cancellation. **Other responses to cancellation include returning the work completed so far, returning an empty result, or returning nil.**"